src/pages/index.js

```diff
@@ -89,31 +134,105 @@ function handleImageClick(data) {
 /*Event Handler*/
 
 function handleProfileEditSubmit(inputValue) {
+  api.updateProfile({
+    name: inputValue.title,
+    description: inputValue.description
+  })
+  .then(updatedUserData => {
     userInfo.setUserInfo({
-        name: inputValue.title,
-        description: inputValue.description,
+      name: updatedUserData.name,
+      about: updatedUserData.about,
     });
     popupWithEditProfileForm.close();
+  })
+  .catch(err => {
+    console.error(err);
+  });
 }
 
 function handleAddCardFormSubmit(inputValue) {
     const cardData = {
         name: inputValue.title,
         link: inputValue.url,
     };
-    renderCard(cardData);
     popupWithAddCardForm.close();
     addCardForm.reset();
+    api.addCard(cardData)
+    .then(newCard => {
+      renderCard(newCard);
+    })
+    .catch(err => {
+      console.error("error adding card:", err);
+    });
+}
+
+function handleDeleteCard(cardId, cardElement) {
+  cardDeleteModal.open();
+
+  cardDeleteModal.setSubmitAction(() => {
+    cardDeleteModal.setIsLoading(true);
+
+    api.deleteCard(cardId)
+      .then(() => {
+        cardElement.remove();
+        cardDeleteModal.close();
+      })
+      .catch((err) => {
+        console.error("Error deleting card:", err);
+      })
+      .finally(() => {
+        cardDeleteModal.setIsLoading(false);
+      });
+  });
+}
+
+
+function handleAvatarEditSubmit(inputValues) {
+  const newAvatarUrl = inputValues.avatar;
+  api.updateAvatar(newAvatarUrl)
+      .then((response) => {
+          console.log("Avatar updated successfully:", response);
+          userInfo.setUserInfo({
+              name: response.name, 
+              description: response.description,
+              avatar: response.avatar,
+          });
+          avatarEditModal.close();
+      })
+      .catch((err) => {
+          console.error("Error updating avatar:", err);
+      });
+}
+
+function handleCardLike(card) {
+  const isLiked = card.isLiked();
+  if (isLiked) {
+    api.dislikeCard(card._id)
+      .then(updatedCard => {
+        card.updateLikes(updatedCard.likes);
+      })
+      .catch(err => console.error(err));
+  } else {
+    api.likeCard(card._id)
+      .then(updatedCard => {
+        card.updateLikes(updatedCard.likes);
+      })
+      .catch(err => console.error(err));
+  }
 }
 
 addCardButton.addEventListener("click", () => {
     popupWithAddCardForm.open();
 });
 
+avatarEditButton.addEventListener("click", () => {
+  avatarEditModal.open();
+});
+
 profileEditButton.addEventListener("click", () => {
     const currentUserInfo = userInfo.getUserInfo();
     profileTitleInput.value = currentUserInfo.name
-    profileDescriptionInput.value = currentUserInfo.description
+    profileDescriptionInput.value = currentUserInfo.about
     popupWithEditProfileForm.open();
 });
 
```

Comment:
`initialCards.forEach((cardData) => renderCard(cardData));
`

It's not needed anymore because `section.renderItems(cards);` renders the cards from the server in line 98